ssz-rs/src/merkleization/mod.rs

```diff
@@ -112,11 +116,15 @@ fn merkleize_chunks_with_virtual_padding(
     let mut hasher = Sha256::new();
     debug_assert!(chunks.len() % BYTES_PER_CHUNK == 0);
     debug_assert!(leaf_count.next_power_of_two() == leaf_count);
+    // this holds as long as usize is no longer than u64
+    debug_assert!((leaf_count.trailing_zeros() as usize) < MAX_MERKLE_TREE_DEPTH);
 
     let height = leaf_count.trailing_zeros() + 1;
 
     if chunk_count == 0 {
         let depth = height - 1;
+        // index is safe as long as depth == leaf_count.trailing_zeros() < MAX_MERKLE_TREE_DEPTH;
+        // qed
```

Comment:
I like this more compact style you had
```suggestion
        // index is safe as long as depth == leaf_count.trailing_zeros() < MAX_MERKLE_TREE_DEPTH; qed
```

Reply:
This was wrapped because of the [max comment width](https://github.com/ralexstokes/ssz-rs/blob/main/rustfmt.toml#L4). We can rephrase it though:
```suggestion
        // index is safe while depth == leaf_count.trailing_zeros() < MAX_MERKLE_TREE_DEPTH; qed
```